frontend/src/tests/instructor-permission-test.js

```diff
@@ -53,54 +206,238 @@ export function instructorsPermissionTests(api) {
+        // we use the default user's admin privilege here to fetch an unrelated instructor
+        let respFetchInst = await apiGET(`/instructor/instructors`);
+        expect(respFetchInst).toHaveStatus("success");
+
+        let respFetchDefaultUser = await apiGET(`/instructor/active_user`);
+        expect(respFetchDefaultUser).toHaveStatus("success");
+        const defaultUser = respFetchDefaultUser.payload;
+
+        const defaultUserWithUpdatedInformation = { ...defaultUser };
+        defaultUserWithUpdatedInformation.email =
+            "instructor_permission_instructor_update_test@test.com";
+
+        // instructor can not modify other instructors
+        // in this example, the instructor only user cant update default user's information
+        await switchToInstructorOnlyUser();
+        let respInvalidRouteWithSession = await apiPOST(
+            `/instructor/instructors`,
+            defaultUserWithUpdatedInformation
+        );
+        expect(respInvalidRouteWithSession).toHaveStatus("error");
+        checkPropTypes(errorPropTypes, respInvalidRouteWithSession);
+
+        await restoreDefaultUser();
+
+        // instructors can modify themselves
+        let respModDefaultUser = await apiPOST(
+            `/instructor/instructors`,
+            defaultUserWithUpdatedInformation
+        );
+        expect(respModDefaultUser).toHaveStatus("success");
+        expect(respModDefaultUser.payload.email).toEqual(
+            "instructor_permission_instructor_update_test@test.com"
+        );
+    });
 
     it("fetch sessions", async () => {
-        let resp = await apiGET("/instructor/instructors");
+        await switchToInstructorOnlyUser();
+        let resp = await apiGET("/instructor/sessions");
+        expect(resp).toHaveStatus("success");
+
+        await restoreDefaultUser();
+        resp = await apiGET("/instructor/sessions");
         expect(resp).toHaveStatus("success");
     });
 
-    it.todo("can't update session");
+    it("can't update session", async () => {
+        await switchToInstructorOnlyUser();
+
+        const newSession = {
+            start_date: "2020-04-10T01:00:00.000Z",
+            end_date: "2020-11-31T00:01:20.000Z",
+            name: "instructor_permission_cant_update_session_test_session",
+        };
+
+        let resp = await apiPOST("/instructor/sessions", newSession);
+        expect(resp).toHaveStatus("error");
+
+        resp = await apiPOST("/admin/sessions", newSession);
+        expect(resp).toHaveStatus("error");
+
+        await restoreDefaultUser();
+    });
 
     it("fetch positions", async () => {
+        await switchToInstructorOnlyUser();
         let resp = await apiGET(`/instructor/sessions/${session.id}/positions`);
         expect(resp).toHaveStatus("success");
+
+        await restoreDefaultUser();
+        resp = await apiGET(`/instructor/sessions/${session.id}/positions`);
+        expect(resp).toHaveStatus("success");
```

Comment:
Are you going to check that the payload contains the positions that are supposed to be there?

Reply:
I would argue checking the content of the response should be checked inside the instructor-tests.js, while intructor-permission-tests.js only checks the accessibility of different routes given a certain set of permissions.